sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -592,6 +592,9 @@ object StructType extends AbstractDataType {
       case (leftUdt: UserDefinedType[_], rightUdt: UserDefinedType[_])
         if leftUdt.userClass == rightUdt.userClass => leftUdt
 
+      case (leftType, rightUdt: UserDefinedType[_])
```

Comment:
Since this is beyond our existing rule, shall we update the function description accordingly?
- https://github.com/apache/spark/pull/26644/files#diff-e9d42ebfb254a463cccff76e8983242cR447-R455

Reply:
I've added a Scaladoc to the private merge function. I think the Javadoc that you're pointing to, is describing the function at a different level. For example, it doesn't mention any UDT's at all. Let me know what you think.